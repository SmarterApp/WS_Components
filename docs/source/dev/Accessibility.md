# Accessibility

## Minimum Requirements
* Use a keyboard to interact with components (tab, space, enter, shift-tab)
* Roles should be used unless input (input type does this automatically)
* Check contrast of components being used

## Resources
* A11y [here](http://a11yproject.com/checklist.html)
* React Accessibility [here](https://reactjs.org/docs/accessibility.html)
* MDN button and links [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_button_role)
* MDN Accessibility [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)

## TSLint
- Linting is enabled for this project to determine if elements are compliant
    * Note this does not check for everything. Manual checks required
- Enable vscode tslint for the best experience when developing