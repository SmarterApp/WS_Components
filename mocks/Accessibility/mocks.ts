import {
  AccResourceGroupModel,
  AccessibilityResourceModel,
  DropDownSelectionModel,
  DropdownProps,
  ItemAccessibilityModalProps
} from "src";
import { action } from "@storybook/addon-actions";

export const accessibilityNotepad: AccessibilityResourceModel = {
  resourceCode: "DigitalNotepad",
  currentSelectionCode: "TDS_SCNotepad",
  order: 1,
  defaultSelection: "TDS_SCNotepad",
  disabled: false,
  description: "Digital Notepad",
  label: "Digital Notepad",
  selections: [
    {
      selectionCode: "TDS_SCNotepad",
      label: "Notepad on",
      order: 1,
      disabled: false,
      hidden: false
    },
    {
      selectionCode: "TDS_SC0",
      label: "Notepad off",
      order: 2,
      disabled: false,
      hidden: false
    },
    {
      selectionCode: "TDS_SC9",
      label: "Notepad disabled",
      order: 3,
      disabled: true,
      hidden: false
    }
  ]
};

export const accessibilityEnglishGlossary: AccessibilityResourceModel = {
  resourceCode: "EnglishGlossary",
  currentSelectionCode: "TDS_WL_0",
  order: 1,
  defaultSelection: "TDS_WL_Glossary",
  disabled: false,
  description: "English Glossary",
  label: "English Glossary",
  selections: [
    {
      selectionCode: "TDS_WL_Glossary",
      label: "English Glossary on",
      order: 1,
      disabled: false,
      hidden: false
    },
    {
      selectionCode: "TDS_WL_0",
      label: "English Glossary off",
      order: 2,
      disabled: false,
      hidden: false
    }
  ]
};

export const mockAccGroup: AccResourceGroupModel = {
  label: "Universal Tools",
  order: 1,
  accessibilityResources: [accessibilityNotepad, accessibilityEnglishGlossary]
};

export const mockAccResourceGroups: AccResourceGroupModel[] = [mockAccGroup];

export const accessibilityModalProp: ItemAccessibilityModalProps = {
  accResourceGroups: [],
  onSave: action("Saved Selections"),
  onReset: action("Reset clicked")
};

export const dropDefaultProp: DropdownProps = {
  disabled: false,
  label: "Notepad",
  selections: accessibilityNotepad.selections,
  defaultSelection: accessibilityNotepad.defaultSelection,
  updateSelection: action("Updated Selection"),
  selectionCode: accessibilityNotepad.currentSelectionCode,
  resourceCode: "Default"
};

export const dropDisabledAllSelectionsProp: DropdownProps = {
  ...dropDefaultProp,
  updateSelection: action("Updated Selection"),
  selections: dropDefaultProp.selections.map(s => {
    return { ...s, disabled: true };
  })
};

export const dropDisabledProp: DropdownProps = {
  ...dropDefaultProp,
  disabled: true,
  updateSelection: action("Updated Selection"),
  selections: dropDefaultProp.selections.map(s => {
    return { ...s, disabled: true };
  })
};
