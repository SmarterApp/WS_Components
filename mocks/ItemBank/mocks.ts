import * as ItemBank from "src/ItemBank/ItemBank"

export const RevisionMockModelOne: ItemBank.RevisionModel = {
    author: "Troy Barnes",
    date: new Date(),
    commitMessage: "Added functionality to the website",
    commitHash: "ab65jg"
}

export const RevisionMockModelTwo: ItemBank.RevisionModel = {
    author: "Pierce Hawthorne",
    date: new Date(),
    commitMessage: "Fixed component",
    commitHash: "h4lso6"
}

export const RevisionMockModelThree: ItemBank.RevisionModel = {
    author: "Annie Edison",
    date: new Date(),
    commitMessage: "Changed one of the pages",
    commitHash: "k5ls58"
}

export const ItemBankPropsMockModel: ItemBank.ItemBankProps = {
    revisions: [
        RevisionMockModelOne,
        RevisionMockModelTwo,
        RevisionMockModelThree
    ]
}