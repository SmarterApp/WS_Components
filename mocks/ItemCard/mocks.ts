import { ItemCardModel } from "@src/index";

export const completeItemCard: ItemCardModel = {
  selected: false,
  bankKey: 187,
  itemKey: 3434,
  interactionTypeCode: "GI",
  interactionTypeLabel: "Item Type",
  title: "ELA Grade 4",
  grade: 4,
  gradeLabel: "Grade 4",
  subjectCode: "ELA",
  subjectLabel: "ELA",
  claimCode: "3",
  claimLabel: "Claim",
  targetDescription: "target description right hereeeee",
  isPerformanceItem: false,
  brailleOnlyItem: false,
  targetId: "dfas",
  depthOfKnowledge: "2",
  commonCoreStandardId: "AB-123",
  ccssDescription: "this is the common core description",
  domain: "domain"
};

export const itemCardList: ItemCardModel[] = [
  {
    bankKey: 187,
    itemKey: 3206,
    title: "Math Grade 6 Claim 1",
    grade: 8,
    gradeLabel: "Grade 6",
    subjectCode: "MATH",
    subjectLabel: "Math",
    claimCode: "MATH1",
    claimLabel: "Concepts and Procedures",
    targetId: "A",
    targetDescription: "Target A description",
    interactionTypeCode: "EQ",
    interactionTypeLabel: "Equation",
    isPerformanceItem: true,
    brailleOnlyItem: false,
    calculator: true
  },
  {
    bankKey: 187,
    itemKey: 3163,
    title: "ELA/ literacy Grade 3 Claim 1",
    grade: 1,
    gradeLabel: "Grade 3",
    subjectCode: "ELA",
    subjectLabel: "ELA/literacy",
    claimCode: "ELA1",
    claimLabel: "Reading",
    targetId: "B",
    targetDescription: "Target B description",
    interactionTypeCode: "MC",
    interactionTypeLabel: "Multiple Choice",
    isPerformanceItem: true,
    brailleOnlyItem: false,
    calculator: false
  },
  {
    bankKey: 187,
    itemKey: 3615,
    title: "Math Grade 6 Claim 4",
    grade: 8,
    gradeLabel: "Grade 6",
    subjectCode: "MATH",
    subjectLabel: "Math",
    claimCode: "MATH4",
    claimLabel: "Modeling/Data Analysis",
    targetId: "C",
    targetDescription: "target C description",
    interactionTypeCode: "MS",
    interactionTypeLabel: "Multi Select",
    isPerformanceItem: false,
    brailleOnlyItem: false
  },
  {
    bankKey: 187,
    itemKey: 2928,
    title: "ELA/ literacy Grade 4 Claim 4",
    grade: 2,
    gradeLabel: "Grade 4",
    subjectCode: "ELA",
    subjectLabel: "ELA/literacy",
    claimCode: "ELA4",
    claimLabel: "Research/Inquiry",
    targetId: "D",
    targetDescription: "target D description",
    interactionTypeCode: "MS",
    interactionTypeLabel: "Multi Select",
    isPerformanceItem: false,
    brailleOnlyItem: false
  }
];

export const sortableItemCards: ItemCardModel[] = [
  {
    bankKey: 187,
    itemKey: 1,
    title: "alpha",
    grade: 5,
    gradeLabel: "Grade 5",
    subjectCode: "MATH",
    subjectLabel: "Math",
    claimCode: "MATH1",
    claimLabel: "Alpha",
    targetId: "A",
    targetDescription: "target A description",
    interactionTypeCode: "EQ",
    interactionTypeLabel: "Equation",
    isPerformanceItem: true,
    brailleOnlyItem: false
  },
  {
    bankKey: 187,
    itemKey: 2,
    title: "bravo",
    grade: 4,
    gradeLabel: "Grade 4",
    subjectCode: "ELA",
    subjectLabel: "ELA/literacy",
    claimCode: "ELA1",
    claimLabel: "Bravo",
    targetId: "B",
    targetDescription: "target B description",
    interactionTypeCode: "MC",
    interactionTypeLabel: "Multiple Choice",
    isPerformanceItem: true,
    brailleOnlyItem: false
  },
  {
    bankKey: 187,
    itemKey: 3,
    title: "charlie",
    grade: 3,
    gradeLabel: "Grade 3",
    subjectCode: "MATH",
    subjectLabel: "Math",
    claimCode: "MATH4",
    claimLabel: "Charlie",
    targetId: "C",
    targetDescription: "target C description",
    interactionTypeCode: "MS",
    interactionTypeLabel: "Multi Select",
    isPerformanceItem: false,
    brailleOnlyItem: false
  },
  {
    bankKey: 187,
    itemKey: 4,
    title: "delta",
    grade: 2,
    gradeLabel: "Grade 2",
    subjectCode: "ELA",
    subjectLabel: "ELA/literacy",
    claimCode: "ELA4",
    claimLabel: "Delta",
    targetId: "D",
    targetDescription: "target D description",
    interactionTypeCode: "MS",
    interactionTypeLabel: "Multi Select",
    isPerformanceItem: false,
    brailleOnlyItem: false
  }
];
