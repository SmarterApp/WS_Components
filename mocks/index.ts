//
// About Item
//
export {
  rubricsWithSamples,
  mockRubrics,
  rubricsEsn,
  aboutItemRevisionMockModel,
  aboutItemMockModel,
  aboutItemMockNoRubric,
  allRubrics
} from "./AboutItem/mocks";

//
// Accessibility
//
export {
  mockAccResourceGroups,
  allAccessibilityResourceGroups
} from "./Accessibility/mocks";

//
// Item Bank
//
export {
  mockBankAboutItemClient,
  mockBankAccessibilityClient,
  mockBankRevisionsClient,
  mockBankSectionsClient,
  itemRevisionMocks,
  sectionMocks
} from "./ItemBank/mocks";

//
// Revisions
//
export { mockRevisions } from "./Revisions/RevisionMocks";
