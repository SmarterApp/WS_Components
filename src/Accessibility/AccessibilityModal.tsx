/// <reference types="google.analytics" />
import * as React from "react";
import {
  AccessibilityResourceModel,
  AccResourceGroupModel,
  ResourceSelectionsModel,
  getResourceTypes
} from "./AccessibilityModels";
import { Dropdown, DropdownProps } from "../index";
import * as ReactModal from "react-modal";

export interface ItemAccessibilityModalProps {
  accResourceGroups: AccResourceGroupModel[];
  onSave(selections: ResourceSelectionsModel): void;
  onReset(): void;
  showModal?: boolean;
}

export interface IsResourceExpanded {
  [resourceType: string]: boolean;
}

export interface ItemAccessibilityModalState {
  resourceTypeExpanded: IsResourceExpanded;
  resourceSelections: ResourceSelectionsModel;
  showModal: boolean;
}

export class ItemAccessibilityModal extends React.Component<
  ItemAccessibilityModalProps,
  ItemAccessibilityModalState
> {
  constructor(props: ItemAccessibilityModalProps) {
    super(props);

    const expandeds: IsResourceExpanded = {};
    const resourceTypes = this.props.accResourceGroups.map(res => {
      return res.label;
    });
    for (const key of resourceTypes) {
      expandeds[key] = false;
    }
    this.state = {
      resourceTypeExpanded: expandeds,
      resourceSelections: {},
      showModal: this.props.showModal || false
    };
  }

  toggleResourceType(resourceType: string) {
    const expandeds = Object.assign({}, this.state.resourceTypeExpanded || {});
    expandeds[resourceType] = !expandeds[resourceType];

    this.setState({
      resourceTypeExpanded: expandeds
    });
  }

  keyboardToggleResourceType(
    e: React.KeyboardEvent<HTMLAnchorElement>,
    resourceType: string
  ): void {
    if (e.keyCode === 23 || e.keyCode === 13) {
      const expandeds = { ...this.state.resourceTypeExpanded };
      expandeds[resourceType] = !expandeds[resourceType];

      this.setState({
        resourceTypeExpanded: expandeds
      });
    }
  }

  // Updates the selection based on user input, if item is braille, streamlined mode needs to be adjusted
  updateSelection = (selectionCode: string, resourceCode: string) => {
    const newSelections = { ...this.state.resourceSelections };
    newSelections[resourceCode] = selectionCode;
    if (resourceCode === "BrailleType") {
      if (selectionCode === "TDS_BT0") {
        newSelections["StreamlinedInterface"] = "TDS_SLM0";
      } else {
        newSelections["StreamlinedInterface"] = "TDS_SLM1";
      }
    }
    this.setState({ resourceSelections: newSelections });
  };

  onSave = (e: React.FormEvent<HTMLFormElement | HTMLButtonElement>) => {
    e.preventDefault();
    this.props.onSave(this.state.resourceSelections || {});
    this.setState({ showModal: false });
  };

  onCancel = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    this.setState({ resourceSelections: {}, showModal: false });
  };

  onReset = (e: React.MouseEvent<HTMLButtonElement>) => {
    this.setState({ resourceSelections: {}, showModal: false });
    this.props.onReset();
  };

  handleShowModal = () => {
    if (!this.state.showModal) {
      ga("send", "event", "button", "OpenAccessibility");
    }

    this.setState({ showModal: true });
  };

  handleHideModal = () => {
    this.setState({ showModal: false });
  };

  renderResourceType(resourceType: string) {
    let resources = this.props.accResourceGroups.filter(
      group => group.label === resourceType
    )[0].accessibilityResources;
    const resourceTypeHeader = (
      <h4 className="green-title">
        <span className="fa fa-tasks" />&nbsp;
        {resourceType}
      </h4>
    );

    const resCount = resources.length;
    const isExpanded = (this.state.resourceTypeExpanded || {})[resourceType];
    if (!isExpanded) {
      resources = resources.slice(0, 4);
    }

    const dropdowns = resources.map(res => {
      const selectedCode =
        (this.state.resourceSelections || {})[res.resourceCode] ||
        res.currentSelectionCode;
      const selections = res.selections.filter(s => !s.hidden);
      const ddProps: DropdownProps = {
        selections,
        defaultSelection: res.currentSelectionCode,
        label: res.label,
        selectionCode: selectedCode,
        disabled: res.disabled,
        updateSelection: this.updateSelection,
        resourceCode: res.resourceCode
      };

      return <Dropdown {...ddProps} key={res.resourceCode} />;
    });

    let expandButton: JSX.Element | undefined;
    if (resCount <= 4) {
      expandButton = undefined;
    } else if (isExpanded) {
      const ariaText = `Display fewer ${resourceType} options`;
      expandButton = (
        <a
          role="button"
          className="expand-button"
          tabIndex={0}
          aria-label={ariaText}
          onClick={() => this.toggleResourceType(resourceType)}
          onKeyUp={e => this.keyboardToggleResourceType(e, resourceType)}
        >
          Show less
        </a>
      );
    } else {
      const ariaText = `Display all ${resourceType} options`;
      expandButton = (
        <a
          role="button"
          className="expand-button"
          tabIndex={0}
          aria-label={ariaText}
          onClick={() => this.toggleResourceType(resourceType)}
          onKeyUp={e => this.keyboardToggleResourceType(e, resourceType)}
        >
          Show all
        </a>
      );
    }

    return (
      <div
        className="accessibility-resource-type section section-light"
        key={resourceType}
      >
        {resourceTypeHeader}
        <div className="accessibility-dropdowns">{dropdowns}</div>
        {expandButton}
      </div>
    );
  }

  render() {
    const types = getResourceTypes(this.props.accResourceGroups);
    const groups = types.map(t => this.renderResourceType(t));

    return (
      <div>
        <button
          className="btn btn-primary"
          onClick={this.handleShowModal}
          aria-label="Open Accessibility Modal"
          tabIndex={0}
        >
          <span className="fa fa-caret-square-o-down" aria-hidden="true" />
          Accessibility
        </button>
        <ReactModal
          isOpen={this.state.showModal}
          contentLabel="Accessibility Modal"
          onRequestClose={this.handleHideModal}
          overlayClassName="react-modal-overlay"
          className="react-modal-content accessibility-modal"
        >
          <div className="modal-wrapper">
            <div className="modal-header">
              <h4 className="modal-title">Accessibility Options</h4>
              <button
                className="close"
                onClick={this.handleHideModal}
                aria-label="Close accessibility options"
              >
                <span className="fa fa-times" aria-hidden="true" />
              </button>
            </div>
            <div className="modal-body">
              <p>
                <span>
                  Options displayed in grey are not available for this item.
                </span>
              </p>
              <form id="accessibility-form" onSubmit={this.onSave}>
                <div className="accessibility-groups">{groups}</div>
              </form>
            </div>
            <div className="modal-footer">
              <button
                className="btn btn-sm btn-primary"
                aria-label="Update options and reload item"
                form="accessibility-form"
                onClick={this.onSave}
              >
                {" "}
                Update
              </button>
              <button
                className="btn btn-sm btn-primary"
                aria-label="Reset all options to default and reload item"
                onClick={this.onReset}
              >
                Reset to Default
              </button>
              <button
                className="btn btn-sm btn-primary"
                aria-label="Cancel and undo changes"
                onClick={this.onCancel}
              >
                Cancel
              </button>
            </div>
          </div>
        </ReactModal>
      </div>
    );
  }
}
