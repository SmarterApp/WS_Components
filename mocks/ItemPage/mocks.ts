import * as H from "history";
import { match } from "react-router";
import {
  ItemViewerContainerProps,
  ItemViewContainerProps,
  ItemModel,
  ItemPageModel,
  ItemIsaapModel,
  AccResourceGroupModel,
  AboutItemModel
} from "src/index";
import { ResourceSelectionsModel } from "src/Accessibility/AccessibilityModels";
import { AboutItemMockModel } from "mocks/AboutItem/mocks";
import { allAccessibilityResourceGroups } from "mocks/Accessibility/mocks";
import { mockPromise } from "mocks/promise";

export const aboutThisClient = (params: ItemModel) =>
  mockPromise(AboutItemMockModel);

export const itemPageClient = (params: ItemModel) =>
  mockPromise<ItemPageModel>(itemPageModelMock);

export const itemAccessibilityClient = (params: ItemIsaapModel) =>
  mockPromise(itemPageMockProps.accResourceGroups);

export const itemPagePath = "/:bankKey:itemKey";

export const itemPageMatch: match<ItemModel> = {
  params: { bankKey: 187, itemKey: 4000 },
  isExact: true,
  path: "itemPagePath",
  url: "/"
};

export const onSave = ((selections: ResourceSelectionsModel) => {}) as ((
  selections: ResourceSelectionsModel
) => void);

export const onReset = () => {};

export const itemPageModelMock: ItemPageModel = {
  itemViewerServiceUrl: "http://ivs.smarterbalanced.org/",
  itemNames: "187-3000",
  brailleItemNames: "187-3000",
  nonBrailleItem: {
    itemName: "187-3000",
    itemKey: 3000,
    bankKey: 187
  },
  brailleItem: {
    itemName: "187-3001",
    itemKey: 3001,
    bankKey: 187
  },
  accessibilityCookieName: "accessibilityCookie",
  isPerformanceItem: false,
  moreLikeThisVM: {
    gradeBelowItems: {
      label: "Grade 5",
      itemCards: []
    },
    sameGradeItems: {
      label: "Grade 6",
      itemCards: []
    },
    gradeAboveItems: {
      label: "Grade 7",
      itemCards: []
    }
  },
  performanceItemDescription: "",
  subject: "ELA",
  brailleItemCodes: [
    "TDS_BT_ECL",
    "TDS_BT_ECN",
    "TDS_BT_EXL",
    "TDS_BT_EXN",
    "TDS_BT_UCL",
    "TDS_BT_UXL"
  ],
  braillePassageCodes: [],
  defaultIsaapCodes: "TDS_ITM1;TDS_APC_SCRUBBER;"
};

export const itemPageMockProps: ItemViewerContainerProps = {
  ...itemPageModelMock,
  onSave,
  onReset,
  showRubrics: true,
  aboutThisItemVM: AboutItemMockModel,
  currentItem: {
    itemName: "187-3000",
    itemKey: 3000,
    bankKey: 187
  },
  accResourceGroups: allAccessibilityResourceGroups
};

export const ItemPageMockPropsNoItem: ItemViewerContainerProps = {
  onSave,
  onReset,
  showRubrics: true,
  aboutThisItemVM: AboutItemMockModel,
  itemViewerServiceUrl: "http://ivs.smarterbalanced.org/",
  itemNames: "",
  brailleItemNames: "",
  nonBrailleItem: {
    itemName: "",
    itemKey: 0,
    bankKey: 0
  },
  brailleItem: {
    itemName: "",
    itemKey: 0,
    bankKey: 0
  },
  accResourceGroups: [] as AccResourceGroupModel[],
  currentItem: {
    itemName: "",
    itemKey: 0,
    bankKey: 0
  },
  accessibilityCookieName: "accessibilityCookie",
  isPerformanceItem: false,
  moreLikeThisVM: {
    gradeBelowItems: {
      label: "Grade 5",
      itemCards: []
    },
    sameGradeItems: {
      label: "Grade 6",
      itemCards: []
    },
    gradeAboveItems: {
      label: "Grade 7",
      itemCards: []
    }
  },
  performanceItemDescription: "",
  subject: "ELA",
  brailleItemCodes: [
    "TDS_BT_ECL",
    "TDS_BT_ECN",
    "TDS_BT_EXL",
    "TDS_BT_EXN",
    "TDS_BT_UCL",
    "TDS_BT_UXL"
  ],
  braillePassageCodes: [],
  defaultIsaapCodes: "TDS_ITM1;TDS_APC_SCRUBBER;"
};
