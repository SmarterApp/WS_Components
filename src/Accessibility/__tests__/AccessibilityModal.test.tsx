import * as React from "react";
import * as ReactDOM from "react-dom";
import { shallow, mount, render } from "enzyme";
import { mockAccResourceGroups } from "mocks/Accessibility/mocks";
import { ItemAccessibilityModal } from "../AccessibilityModal";
import { ResourceSelectionsModel } from "../AccessibilityModels";

describe("AccessibilityModal", () => {
  const onSaveHandler = jest.fn((selections: ResourceSelectionsModel) => {
    return;
  });
  const onResetHandler = jest.fn();

  const wrapper = shallow(
    <ItemAccessibilityModal
      accResourceGroups={mockAccResourceGroups}
      onSave={onSaveHandler}
      onReset={onResetHandler}
    />
  );

  it("defualt render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("accessibility open", () => {
    wrapper.setProps({ showModal: true });
    expect(wrapper).toMatchSnapshot();
  });
});
