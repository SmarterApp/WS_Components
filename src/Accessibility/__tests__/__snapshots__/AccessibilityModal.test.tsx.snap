// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessibilityModal accessibility open 1`] = `
<div>
  <button
    aria-label="Open Accessibility Modal"
    className="btn btn-primary"
    onClick={[Function]}
    tabIndex={0}
  >
    <span
      aria-hidden="true"
      className="fa fa-caret-square-o-down"
    />
    Accessibility
  </button>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="react-modal-content accessibility-modal"
    closeTimeoutMS={0}
    contentLabel="Accessibility Modal"
    isOpen={false}
    onRequestClose={[Function]}
    overlayClassName="react-modal-overlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      className="modal-wrapper"
    >
      <div
        className="modal-header"
      >
        <h4
          className="modal-title"
        >
          Accessibility Options
        </h4>
        <button
          aria-label="Close accessibility options"
          className="close"
          onClick={[Function]}
        >
          <span
            aria-hidden="true"
            className="fa fa-times"
          />
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p>
          <span>
            Options displayed in grey are not available for this item.
          </span>
        </p>
        <form
          id="accessibility-form"
          onSubmit={[Function]}
        >
          <div
            className="accessibility-groups"
          >
            <div
              className="accessibility-resource-type section section-light"
              key="Universal Tools"
            >
              <h4
                className="green-title"
              >
                <span
                  aria-hidden="true"
                  className="fa fa-tasks"
                />
                 
                Universal Tools
              </h4>
              <div
                className="accessibility-dropdowns"
              >
                <Dropdown
                  defaultSelection="TDS_SCNotepad"
                  disabled={false}
                  key="DigitalNotepad"
                  label="Digital Notepad"
                  resourceCode="DigitalNotepad"
                  selectionCode="TDS_SCNotepad"
                  selections={
                    Array [
                      Object {
                        "disabled": false,
                        "hidden": false,
                        "label": "Notepad on",
                        "order": 1,
                        "selectionCode": "TDS_SCNotepad",
                      },
                      Object {
                        "disabled": false,
                        "hidden": false,
                        "label": "Notepad off",
                        "order": 2,
                        "selectionCode": "TDS_SC0",
                      },
                      Object {
                        "disabled": true,
                        "hidden": false,
                        "label": "Notepad disabled",
                        "order": 3,
                        "selectionCode": "TDS_SC9",
                      },
                    ]
                  }
                  updateSelection={[Function]}
                />
                <Dropdown
                  defaultSelection="TDS_WL_0"
                  disabled={false}
                  key="EnglishGlossary"
                  label="English Glossary"
                  resourceCode="EnglishGlossary"
                  selectionCode="TDS_WL_0"
                  selections={
                    Array [
                      Object {
                        "disabled": false,
                        "hidden": false,
                        "label": "English Glossary on",
                        "order": 1,
                        "selectionCode": "TDS_WL_Glossary",
                      },
                      Object {
                        "disabled": false,
                        "hidden": false,
                        "label": "English Glossary off",
                        "order": 2,
                        "selectionCode": "TDS_WL_0",
                      },
                    ]
                  }
                  updateSelection={[Function]}
                />
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        className="modal-footer"
      >
        <button
          aria-label="Update options and reload item"
          className="btn btn-sm btn-primary"
          form="accessibility-form"
          onClick={[Function]}
        >
           
          Update
        </button>
        <button
          aria-label="Reset all options to default and reload item"
          className="btn btn-sm btn-primary"
          onClick={[Function]}
        >
          Reset to Default
        </button>
        <button
          aria-label="Cancel and undo changes"
          className="btn btn-sm btn-primary"
          onClick={[Function]}
        >
          Cancel
        </button>
      </div>
    </div>
  </Modal>
</div>
`;

exports[`AccessibilityModal defualt render 1`] = `
<div>
  <button
    aria-label="Open Accessibility Modal"
    className="btn btn-primary"
    onClick={[Function]}
    tabIndex={0}
  >
    <span
      aria-hidden="true"
      className="fa fa-caret-square-o-down"
    />
    Accessibility
  </button>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="react-modal-content accessibility-modal"
    closeTimeoutMS={0}
    contentLabel="Accessibility Modal"
    isOpen={false}
    onRequestClose={[Function]}
    overlayClassName="react-modal-overlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      className="modal-wrapper"
    >
      <div
        className="modal-header"
      >
        <h4
          className="modal-title"
        >
          Accessibility Options
        </h4>
        <button
          aria-label="Close accessibility options"
          className="close"
          onClick={[Function]}
        >
          <span
            aria-hidden="true"
            className="fa fa-times"
          />
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p>
          <span>
            Options displayed in grey are not available for this item.
          </span>
        </p>
        <form
          id="accessibility-form"
          onSubmit={[Function]}
        >
          <div
            className="accessibility-groups"
          >
            <div
              className="accessibility-resource-type section section-light"
              key="Universal Tools"
            >
              <h4
                className="green-title"
              >
                <span
                  aria-hidden="true"
                  className="fa fa-tasks"
                />
                 
                Universal Tools
              </h4>
              <div
                className="accessibility-dropdowns"
              >
                <Dropdown
                  defaultSelection="TDS_SCNotepad"
                  disabled={false}
                  key="DigitalNotepad"
                  label="Digital Notepad"
                  resourceCode="DigitalNotepad"
                  selectionCode="TDS_SCNotepad"
                  selections={
                    Array [
                      Object {
                        "disabled": false,
                        "hidden": false,
                        "label": "Notepad on",
                        "order": 1,
                        "selectionCode": "TDS_SCNotepad",
                      },
                      Object {
                        "disabled": false,
                        "hidden": false,
                        "label": "Notepad off",
                        "order": 2,
                        "selectionCode": "TDS_SC0",
                      },
                      Object {
                        "disabled": true,
                        "hidden": false,
                        "label": "Notepad disabled",
                        "order": 3,
                        "selectionCode": "TDS_SC9",
                      },
                    ]
                  }
                  updateSelection={[Function]}
                />
                <Dropdown
                  defaultSelection="TDS_WL_0"
                  disabled={false}
                  key="EnglishGlossary"
                  label="English Glossary"
                  resourceCode="EnglishGlossary"
                  selectionCode="TDS_WL_0"
                  selections={
                    Array [
                      Object {
                        "disabled": false,
                        "hidden": false,
                        "label": "English Glossary on",
                        "order": 1,
                        "selectionCode": "TDS_WL_Glossary",
                      },
                      Object {
                        "disabled": false,
                        "hidden": false,
                        "label": "English Glossary off",
                        "order": 2,
                        "selectionCode": "TDS_WL_0",
                      },
                    ]
                  }
                  updateSelection={[Function]}
                />
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        className="modal-footer"
      >
        <button
          aria-label="Update options and reload item"
          className="btn btn-sm btn-primary"
          form="accessibility-form"
          onClick={[Function]}
        >
           
          Update
        </button>
        <button
          aria-label="Reset all options to default and reload item"
          className="btn btn-sm btn-primary"
          onClick={[Function]}
        >
          Reset to Default
        </button>
        <button
          aria-label="Cancel and undo changes"
          className="btn btn-sm btn-primary"
          onClick={[Function]}
        >
          Cancel
        </button>
      </div>
    </div>
  </Modal>
</div>
`;
